internal/extension/smartagentextension/config.go

```diff
@@ -93,7 +93,19 @@ func customUnmarshaller(componentViperSection *viper.Viper, intoCfg interface{})
 		return fmt.Errorf("failed setting collectd config defaults: %w", err)
 	}
 
+	// The default on CollectdConfig is 0, use the default if this is the case.
+	if collectdConfig.IntervalSeconds == 0 {
+		collectdConfig.IntervalSeconds = defaultIntervalSeconds
+	}
+
+	// If the HasGenericJMXMonitor is unset and not user specified, set it to true.
+	if !collectdConfig.HasGenericJMXMonitor && collectdSettings["hasgenericjmxmonitor"] == nil {
```

Comment:
If `hasgenericjmxmonitor` were in the provided config I believe the unmarshal strict would have errored since the yaml tag is set to omit.  I'm not 100% this field is necessary for us to worry about and think the collectd manager maintains it as genericjmx monitors call their `Configure()` methods.

Reply:
at any rate, we can address it as a genericjmx integration test is added

Reply:
👍

Reply:
Mind removing this block for the PR?

Reply:
Missed this comment initially. Updated.